### PR TITLE
Adding AppLogger class and examples

### PR DESCRIPTION
**Pull request by anandchugh**
Adding Python logger class using Opencensus SDK with examples and documentation.

### CODE REVIEW

File: azure_monitor/python_logger_opencensus_azure/README.md [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+
+    `get_tracer` function return a Tracer object. It sets the [component_name][6] to `ai.cloud.role` which is used to identify component in application map. It also sets parent_tracer for proper correlation.
+
+* **Function `enable_flask`**
+
+    ```python
+    def enable_flask(self,flask_app,component_name="AppLogger"):
+    ```
+
+    `enable_flask` function enables Flask middleware as mentioned in the [documentation.][5] This is required to enable log tracing for flask applications.
```
Comment (lzchen) [inline azure_monitor/python_logger_opencensus_azure/README.md:159]:
```suggestion
    `enable_flask` function enables Flask middleware as mentioned in the [documentation.][5] This is required to enable tracing for flask applications.
```

File: azure_monitor/python_logger_opencensus_azure/README.md [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+* Add disabled logger which can be used during Unit testing.
+
+Assuming, if we need to add code for above scenarios, then this will get repeated for every python project in our application. To avoid repetition and hence enabling code reuse, this python logger class is required. 
+
+## Prerequisites for using AppLogger
+
+* An Azure subscription
+* Create an [Application Insights resource](https://docs.microsoft.com/en-us/azure/azure-monitor/app/create-new-resource) and store the application insights instrumentation key
+ which is used in python logger class.
+* Python installation above 3.7.0 version.
```
Comment (lzchen) [inline azure_monitor/python_logger_opencensus_azure/README.md:78]: Is `> Python 3.7.0` a must?

File: azure_monitor/python_logger_opencensus_azure/README.md [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+        config={"app_insights_key"="<some_key>"}
+        app_logger= AppLogger(config)
+        tracer= app_logger.get_tracer(component_name="main")
+        # call package1 function
+        some_result = package1_function(app_logger=app_logger,parent_tracer= tracer)
+    
+        # package1
+        config={"app_insights_key"="<some_key>"}
+        def package1_function(app_logger, parent_tracer=None):
+            app_logger = AppLogger(config)
```
Comment (lzchen) [inline azure_monitor/python_logger_opencensus_azure/README.md:187]: Should use the passed in `app_logger` correct?

File: azure_monitor/python_logger_opencensus_azure/README.md [hunk 4]
```diff
@@ -0,0 +1,310 @@
+# Overview
+
+This document covers the python logger class using [OpenCensus-Python sdk][1] for [Azure Application Insights][3].
```
Comment (lzchen) [inline azure_monitor/python_logger_opencensus_azure/README.md:3]: `Azure Application Insights` link does not link to anything.

File: azure_monitor/python_logger_opencensus_azure/monitoring/examples/client.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+app_logger = AppLogger(config=logging_config)
+logger = app_logger.get_logger(component_name=component_name)
+tracer = app_logger.get_tracer(component_name=component_name)
+
+def call_api_3():
+    """Function calling endpoint of API3
+    """
+    logger.info("Calling api 3")
+    response = requests.get(url='http://localhost:8300/')
```
Comment (lzchen) [inline azure_monitor/python_logger_opencensus_azure/monitoring/examples/client.py:27]: Do we want to use `opencensus-ext-requests` as well to track this?

File: azure_monitor/python_logger_opencensus_azure/monitoring/src/logger.py [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+        app_insights_cs = "InstrumentationKey=" + self._get_app_insights_key()
+        log_handler = AzureLogHandler(
+            connection_string=app_insights_cs, export_interval=0.0
+        )
+        log_handler.add_telemetry_processor(self._get_callback(component_name))
+        log_handler.name = self.HANDLER_NAME
+        log_handler.addFilter(CustomDimensionsFilter(custom_dimensions))
+        return log_handler
+
+    def get_log_exporter(self, component_name="AppLogger"):
```
Comment (lzchen) [inline azure_monitor/python_logger_opencensus_azure/monitoring/src/logger.py:69]:
```suggestion
    def get_trace_exporter(self, component_name="AppLogger"):
```

File: azure_monitor/python_logger_opencensus_azure/monitoring/src/logger.py [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+        """[Get log exporter]
+
+        Returns:
+            [AzureExporter]: [Azure Log Exporter]
```
Comment (lzchen) [inline azure_monitor/python_logger_opencensus_azure/monitoring/src/logger.py:73]:
```suggestion
            [AzureExporter]: [Azure Trace Exporter]
```

File: azure_monitor/python_logger_opencensus_azure/monitoring/src/logger.py (hunk 6 above)
Comment (anandchugh) [reply azure_monitor/python_logger_opencensus_azure/monitoring/src/logger.py:69]: Thanks for fixing this.

**Comment by anandchugh** [comment]
Thanks @lzchen  for reviewing the code and fixing issues also.

**Pull request merged**